packages/process/src/node/process-manager.ts

```diff
@@ -4,35 +4,80 @@
  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
-import { injectable } from 'inversify';
+import { injectable, inject } from 'inversify';
 import { Process } from './process';
 import { Emitter, Event } from '@theia/core/lib/common';
+import { ILogger } from '@theia/core/lib/common/logger';
+import { BackendApplicationContribution } from '@theia/core/lib/node';
 
 @injectable()
-export class ProcessManager {
+export class ProcessManager implements BackendApplicationContribution {
 
-    protected readonly processes: Map<number, Process> = new Map();
     protected id: number = 0;
-    protected readonly deleteEmitter = new Emitter<number>();
+    protected readonly processes: Map<number, Process>;
+    protected readonly deleteEmitter: Emitter<number>;
 
+    constructor( @inject(ILogger) protected logger: ILogger) {
+        this.processes = new Map();
+        this.deleteEmitter = new Emitter<number>();
+    }
+
+    /**
+     * Registers the given process into this manager. Both on process termination and on error,
+     * the process will be automatically removed from the manager.
+     *
+     * @param process the process to register.
+     */
     register(process: Process): number {
         const id = this.id;
         this.processes.set(id, process);
+        process.onExit(() => this.unregister(process));
+        process.onError(() => this.unregister(process));
         this.id++;
         return id;
     }
 
-    get(id: number): Process | undefined {
-        return this.processes.get(id);
+    /**
+     * Removes the process from this process manager. Invoking this method, will make
+     * sure that the process is terminated before eliminating it from the manager's cache.
+     *
+     * @param process the process to unregister from this process manager.
+     */
+    protected unregister(process: Process): void {
+        const processLabel = this.getProcessLabel(process);
+        this.logger.info(`Unregistering process. ${processLabel}`);
```

Comment:
I think info is a big much here ? debug seems more appropriate.
The user will get notified that the process is killed at the info level, however I don't think the user cares about the internal registry.

Reply:
Same with file watching, right? We use `info` for file-watching. I disagree with you, but I want this PR to get merged. I will change it to debug as you have requested.

Reply:
File watching is user visible however ? while this is not... thx for the change.